storage/dataflux/fast_list.go

```diff
@@ -150,6 +179,39 @@ func (c *Lister) NextBatch(ctx context.Context) ([]*storage.ObjectAttrs, error)
 
 // Close closes the range channel of the Lister.
 func (c *Lister) Close() {
+	if c.ranges != nil {
+		close(c.ranges)
+	}
+}
 
-	// TODO: Close range channel for worksteal lister.
+// updateStartEndOffset updates start and end offset based on prefix.
+// If a prefix is given, start and end should contain the value after prefix.
+// For example, if start is "prefix_a", then start should be "_a".
+func updateStartEndOffset(start, end, prefix string) (string, string) {
+	if prefix == "" {
+		return start, end
+	}
+	if start != "" && end != "" && start >= end {
```

Comment:
Shouldn't this possibly modify `start` so that the prefix is not included if present?

Reply:
If start comes lexicographically after end  or equal to end, then no objects will be listed. Adding prefix or not,  will not make a difference.